src/db.rs

```diff
@@ -23,19 +21,22 @@ pub struct DB {
 impl ReadonlyStorage for DB {
     fn get(&self, key: &[u8]) -> Option<Vec<u8>> {
         let buf = Buffer::from_vec(key.to_vec());
-        // TODO: dynamic size (https://github.com/CosmWasm/go-cosmwasm/issues/59)
-        let mut result_buf = Buffer::with_capacity(DB_GET_MAX_RESULT_LENGTH);
-        let res = (self.vtable.read_db)(self.state, buf, result_buf);
+        let mut result_buf = Buffer::default();
+        let res = (self.vtable.read_db)(self.state, buf, &mut result_buf as *mut Buffer);
 
-        // read in the number of bytes returned
+        // check if the Go implementation panicked
         if res < 0 {
-            // TODO
-            panic!("val was not big enough for data");
+            // TODO handle this better
```

Comment:
oh, cool! should i pull in (or rebase on) that branch?

Reply:
I would merge this first. There are MANY breaking changes in 0.8, so I would like to test these memory changes on a stable platform, before making that big update.

Reply:
Oh, nevermind, you're talking about another repo :)
Should i change the traits here as well?

Reply:
No. That is for later. This will all update later.
Let's do one thing at a time. It was just a heads up.

(If you want to participate in the update, you are welcome to, I would like to finalize and merge this PR before starting it. And make a few last changes to the CosmWasm API)

Reply:
Stabilizing cosmwasm with robust APIs and interfaces is a priority for us, so it's likely i will help get this across the finishing line over the next few days and weeks